script.js

```diff
@@ -1,4 +1,259 @@
+      if (arr[j].score > arr[j + 1].score) {
+        // If the condition is true then swap them
+        var temp = arr[j];
+        arr[j] = arr[j + 1];
+        arr[j + 1] = temp;
+      }
+    }
+  }
+}
+
+// BubbleSort directly to object.score highest to lowest (tbd)
+function bblSortDice(arr) {
+  for (var i = 0; i < arr.length; i++) {
+    // Last i elements are already in place
+    for (var j = 0; j < arr.length - i - 1; j++) {
+      // Checking if the item at present iteration
+      // is greater than the next iteration
+      if (arr[j].combinedDiceNumber > arr[j + 1].combinedDiceNumber) {
+        // If the condition is true then swap them
+        var temp = arr[j];
+        arr[j] = arr[j + 1];
+        arr[j + 1] = temp;
+      }
+    }
+  }
+}
+
+// Sort lowest to highest
+function bblSortNumberLow(arr) {
+  for (var i = 0; i < arr.length; i++) {
+    // Last i elements are already in place
+    for (var j = 0; j < arr.length - i - 1; j++) {
+      // Checking if the item at present iteration
+      // is greater than the next iteration
+      if (arr[j] > arr[j + 1]) {
+        // If the condition is true then swap them
+        var temp = arr[j];
+        arr[j] = arr[j + 1];
+        arr[j + 1] = temp;
+      }
+    }
+  }
+  // Print the sorted array
+  return arr;
+}
+
+// Sort array highest to lowest
+function bblSortNumberHigh(array) {
+  for (i = 0; i < array.length - 1; i++) {
+    for (j = 0; j < array.length - i - 1; j++) {
+      if (array[j] < array[j + 1]) {
+        var temp = array[j];
+        array[j] = array[j + 1];
+        array[j + 1] = temp;
+      }
+    }
+  }
+  return array;
+}
+
+function getCombinedDiceNumber(dice_arr, getHighLowOption) {
+  var combinedDiceNumber = "";
+  if (getHighLowOption == "low") {
+    dice_arr = bblSortNumberLow(dice_arr);
+  } else {
+    dice_arr = bblSortNumberHigh(dice_arr);
+  }
+  for (let i = 0; i < dice_arr.length; i++) {
+    combinedDiceNumber = combinedDiceNumber + dice_arr[i];
+  }
+  return combinedDiceNumber;
+}
+
+function rollDice() {
+  // produces a decimal between 0 and 6
+  var randomDecimal = Math.random() * 6;
+
+  // take off the decimal
+  var randomInteger = Math.floor(randomDecimal);
+
+  // it's anumber from 0 - 5 ... add 1
+  var diceNumber = randomInteger + 1;
+
+  return diceNumber.toString();
+}
+
+function roll_n_Dices(numRolls) {
+  let diceArray = [];
+  for (var i = 0; i < numRolls; i++) {
+    diceArray.push(rollDice());
+  }
+  return diceArray;
+}
+
+function playersCreation(numOfPlayers) {
+  for (let i = 0; i < numOfPlayers; i++) {
+    //    plist.push(new player("p" + i, bblSortNumber(roll_n_Dices(5)), rollDice()));   // For testing purpose, roll random 5 dices
+    plist.push(new player("p" + i, [], 0, 0));
+  }
+}
+
+function printScoreBoard(playerlists, mode) {
+  var output = "The score board now!<br>";
+  bblSort(plist); //sort the player list according to score lowest first
+  console.log(playerlists);

```

Comment:
good use of console.log statements for debugging but they are usually removed before shipping to production in a professional setting (the equivalent of submitting your project for this course)

applies for your other console.log statements too